File: solution/1395. Count Number of Teams/1395_02.cpp
class Solution {
public:
    int numTeams(vector<int>& rating) {
        int n = rating.size();
        
        int res = 0;
        for (int i = 1; i < n - 1; ++i) {
            int leftSmaller = 0, leftGreater = 0, rightSmaller = 0, rightGreater = 0;
            for (int j = 0; j < n; ++j) {
                if (rating[i] < rating[j]) {
                    if (i < j)
                        rightGreater++;
                    else
                        leftGreater++;
                }
                if (rating[j] < rating[i]) {
                    if (j < i)
                        leftSmaller++;
                    else
                        rightSmaller++;
                }
            }
            res += leftSmaller * rightGreater + leftGreater * rightSmaller;
        }
        return res;
    }
};

// x x x x x x x x
//     j   i   j

File: solution/1395. Count Number of Teams/1395_02.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        n = len(rating)
        
        res = 0
        for i in range(1, n - 1):
            left_smaller = left_greater = right_smaller = right_greater = 0
            for j in range(n):
                if rating[j] < rating[i]:
                    if j < i:
                        left_smaller += 1
                    else:
                        right_smaller += 1
                if rating[i] < rating[j]:
                    if j < i:
                        left_greater += 1
                    else:
                        right_greater += 1
            res += left_smaller * right_greater + left_greater * right_smaller
        return res
    

# x x x x x x x x
#     j   i   j
